components/api-server/src/bin/api_server.rs

```diff
@@ -0,0 +1,100 @@
+use axum::{
+    Json,
+    Router,
+    extract::{Path, State},
+    http::StatusCode,
+    response::{
+        IntoResponse,
+        Sse,
+        sse::{Event, KeepAlive},
+    },
+    routing::{get, post},
+};
+use clap::Parser;
+use clp_rust_utils::package_config;
+use futures::{Stream, StreamExt};
+use thiserror::Error;
+use tracing_subscriber::{self, prelude::*};
+
+#[derive(Parser)]
+#[command(version, about)]
+struct Args {
+    #[arg(long)]
+    package_root: String,
+}
+
+#[tokio::main]
+async fn main() -> anyhow::Result<()> {
+    tracing_subscriber::registry()
+        .with(tracing_subscriber::fmt::layer())
+        .with(tracing_subscriber::EnvFilter::from_default_env())
+        .init();
+
+    let args = Args::parse();
+    let package_root = std::path::Path::new(&args.package_root);
+
+    let config_path = package_root.join(package_config::DEFAULT_CONFIG_FILE_PATH);
+    let config_string = std::fs::read_to_string(config_path)?;
+    let config: package_config::Config = serde_yaml::from_str(&config_string)?;
+
+    let credentials_path = package_root.join(package_config::DEFAULT_CREDENTIALS_FILE_PATH);
+    let credentials_string = std::fs::read_to_string(credentials_path)?;
+    let credentials: package_config::Credentials = serde_yaml::from_str(&credentials_string)?;
+
+    let client = clp_client::Client::connect(&config, &credentials).await?;
+
+    let addr = format!("{}:{}", &config.api_server.host, &config.api_server.port);
+    let listener = tokio::net::TcpListener::bind(&addr).await?;
+    tracing::event!(tracing::Level::INFO, "Server started at {addr}");
+
+    let app = Router::new()
+        .route("/", get(root))
+        .route("/query", post(query))
+        .route("/query_results/{search_job_id}", get(query_results))
+        .with_state(client);
+    axum::serve(listener, app).await?;
+    Ok(())
+}
+
+async fn root() -> String {
+    "API server is running".to_owned()
```

Comment:
shall we (also?) bind this to a `/health` path? the `/health` path seems more conventional for checking health of a server

for the background, we need to do healthcheck in Docker Compose / K8s for automatic healing